adoptopenjdk-api-v3-frontend/src/test/kotlin/net/adoptopenjdk/api/AssetsResourceReleaseNamePathTest.kt

```diff
@@ -0,0 +1,91 @@
+import net.adoptopenjdk.api.v3.dataSources.APIDataStore
+import net.adoptopenjdk.api.v3.dataSources.SortMethod
+import net.adoptopenjdk.api.v3.dataSources.SortOrder
+import net.adoptopenjdk.api.v3.filters.ReleaseFilter
+import net.adoptopenjdk.api.v3.models.Release
+import net.adoptopenjdk.api.v3.models.ReleaseType
+import net.adoptopenjdk.api.v3.models.Vendor
+import org.hamcrest.Description
+import org.hamcrest.TypeSafeMatcher
+import org.junit.jupiter.api.DynamicTest
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestFactory
+import org.junit.jupiter.api.extension.ExtendWith
+import java.util.stream.Stream
+
+@ExtendWith(value = [DbExtension::class])
+@QuarkusTest
+class AssetsResourceReleaseNamePathTest : FrontendTest() {
+    @TestFactory
+    fun filtersByReleaseNameCorrectly(): Stream<DynamicTest> {
+        return Vendor
+            .values()
+            .flatMap { vendor ->
+                APIDataStore
+                    .getAdoptRepos()
+                    .allReleases
+                    .getReleases(ReleaseFilter(vendor = vendor), SortOrder.DESC, SortMethod.DEFAULT)
+                    .take(3)
+                    .flatMap { release ->
+                        ReleaseType
+                            .values()
+                            .map { "/v3/assets/release_name/$vendor/${release.release_name}" }
+                            .map {
+                                DynamicTest.dynamicTest(it) {
+                                    RestAssured.given()
+                                        .`when`()
+                                        .get(it)
+                                        .then()
+                                        .statusCode(200)
+                                        .and()
+                                        .body(object : TypeSafeMatcher<String>() {
+                                            override fun describeTo(description: Description?) {
+                                                description!!.appendText("json")
+                                            }
+
+                                            override fun matchesSafely(p0: String?): Boolean {
+                                                val returnedRelease = JsonMapper.mapper.readValue(p0, Release::class.java)
+                                                return returnedRelease.id == release.id
+                                            }
+                                        })
+                                }
+                            }
+                            .asSequence()
+                    }
+                    .asIterable()
+            }
+            .stream()
+    }
+
+    @Test
+    fun `non-existent release name 404s`() {
+        RestAssured.given()
+            .`when`()
+            .get("/v3/assets/release_name/adoptopenjdk/foo")
+            .then()
+            .statusCode(404)
+    }
+
+    @Test
+    fun `release with different vendor 404s`() {
```

Comment:
I was kind of expecting this to be asserting a HTTP 500 when he have the same release for different vendors but the test is passing, maybe something's not quite right here in the test setup?

Reply:
I think 404 is correct, you are asking for a release with a given name and vendor, but that combination does not exist, I think not found is correct for that scenario

Reply:
ah ok, sorry getting confused with the case where we serve a 500 when the data is bad